File: python3/longest_continuous_increasing_subsequence.py
# Space : O(1)
# Time  : O(n)

class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        if not nums:
            return 0

        s, ans = 0, 1
        n = len(nums)

        for i in range(1, n):
            if nums[i] > nums[i-1]:
                ans = max(ans, i-s+1)
            else:
                s = i

        return ans

File: golang/longest_continuous_increasing_subsequence.go
package main

// Space : O(1)
// Time  : O(n)

func findLengthOfLCIS(nums []int) int {
    n := len(nums)
    if n == 0 {
        return 0
    }
    
    s, ans := 0, 1
    
    for i := 1; i < n; i++ {
        if nums[i] > nums[i-1] {
            ans = Max(ans, i-s+1)
        } else {
            s = i
        }
    }
    
    return ans
}